### ISSUE TITLE
Can't use arrow keys to edit text

### ISSUE THREAD

**Issue by Leif-W** [Telegram-FOSS-Team/Telegram-FOSS#618]
I've used Hacker's Keyboard specifically to allow for ease of use and accessibility as a disabled person for this task in many dozens of other apps over the past 8 years, since I first started using Android.  Telegram-FOSS is the first app I have encountered that does not handle arrow key codes properly.  Instead of the expected, intuitive, normative, and sane action of very simply moving the cursor left, right, up, or down when inputing text, the cursor just leaves the text editing box entirely, though the virtual keyboard remains open, yet no further text can be entered until the touch screen is used to return the focus to the text area input field.  I do not know where the focus goes, if it was intentionally designed to do this and if so, what mistaken rationality is used to justify an incorrect, user-unfriendly, unexpected action.

**Comment by Barkuti** [comment]
Yes, this is a problem. I am also a user of Telegram FOSS and Hacker's Keyboard, and noticed this bug since version 8.7.4: cursor keys cause a change of context out of the input field. I was hoping for it to be a momentary bug, and so I just downgraded to v8.6.2 a little over two months ago to keep using my cursor keys properly. In Telegram FOSS 8.7.4 and 8.8.5 the up cursor key keeps selecting chat balloons, and the other keys jump to other application fields; it's like if the cursor keys had some strange modifier key stuck enabled. However, when modifier keys are used along the cursor keys they work correctly: I can hold **Shift** and select text, or hold **Ctrl** and jump between words. This is proof that this is a bug, and was not intended to happen.
Of course this must be fixed as soon as possible, so please fix it. The only workaround for now is to keep using good 'ol v8.6.2. Downgrading for now.

**Comment by Barkuti** [comment]
Just installed non-FOSS official Telegram from [https://telegram.org/android](https://telegram.org/android), to check where the goof up comes from; the version I got is 8.9.3, and the arrow keys bug is there. 😒

**Comment by Barkuti** [comment]
Bug is still present in non-FOSS Telegram 9.3.2, tested it personally. 🙄
I opened a new related thread in bugs.telegram.org in order to better document this and make it more obvious: [Cursor keys / arrow keys NOT working properly](https://bugs.telegram.org/c/23671).
Can't say that I like this situation, so I may have to do something else to give the developers or whoever @#$% is giving them orders a nice slap on the wrist.
In the meantime you can upvote my newly opened thread there; that will certainly do more than nothing. Thanks.

**Comment by Barkuti** [comment]
Hello. 😃
Just dropped by to say that, according to Telegram notifications and related threads like [Cursor keys / arrow keys NOT working properly](https://bugs.telegram.org/c/23671) in the awkward official Telegram forum, the bug is fixed since version 9.5.2.
